### PR TITLE
Add uniqueness check for added rights in createRoleValidator, repair …

### PR DESCRIPTION
**Pull request by ndoriomedov**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/846

### CODE REVIEW

File: test/RightsService.Validation.UnitTests/CreateRoleRequestValidatorTests.cs [hunk 1]
```diff
@@ -3,94 +3,140 @@
...
+            _autoMocker
+                .Setup<IRightRepository, List<DbRight>>(x => x.GetRightsList())
+                .Returns(_dbRightsList);
+
+            _autoMocker
+                .Setup<IMemoryCache, ICacheEntry>(x => x.CreateEntry(It.IsAny<object>()))
+                .Returns(_autoMocker.GetMock<ICacheEntry>().Object);
+
+            _validator = new CreateRoleRequestValidator(
+                _autoMocker.GetMock<IRightRepository>().Object,
```
Comment (Cheephot) [inline test/RightsService.Validation.UnitTests/CreateRoleRequestValidatorTests.cs:68]: May be Use CreateInstance?

**Review by Cheephot — APPROVED** [review]

File: src/RightsService.Validation/CreateRoleRequestValidator.cs [hunk 2]
```diff
@@ -11,11 +12,13 @@ namespace LT.DigitalOffice.RightsService.Validation
...
         public CreateRoleRequestValidator(
-            IRightRepository repository,
+            IRightRepository rightRepository,
+            IRoleRepository roleRepository,
+            IRoleRightsCompareHelper compareHelper,
             IMemoryCache cache)
         {
             RuleFor(x => x.Name)
-            .NotEmpty().WithMessage("Role name must not be empty.");
+                .NotEmpty().WithMessage("Role name must not be empty.");
```
Comment (BelehovEgor) [inline src/RightsService.Validation/CreateRoleRequestValidator.cs:21]: move message to next line

File: src/RightsService.Validation/Helpers/Interfaces/IRoleRightsCompareHelper.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+using LT.DigitalOffice.RightsService.Data.Interfaces;
+using System;
+using System.Collections.Generic;
+
+namespace LT.DigitalOffice.RightsService.Validation.Helpers.Interfaces
+{
+    [AutoInject]
+    public interface IRoleRightsCompareHelper
+    {
+        public bool Compare(List<int> addedRights, IRoleRepository repository);
```
Comment (BelehovEgor) [inline src/RightsService.Validation/Helpers/Interfaces/IRoleRightsCompareHelper.cs:11]: i guess better inject repository

File: src/RightsService.Validation/Helpers/RoleRightsCompareHelper.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    public class RoleRightsCompareHelper : IRoleRightsCompareHelper
+    {
+        public bool Compare(List<int> addedRights, IRoleRepository repository)
+        {
+            IEnumerable<DbRole> roles = repository.GetAll();
+            HashSet<int> hashSet = new(addedRights);
+
+            foreach(DbRole role in roles)
+            {
+                List<int> rights = new();
```
Comment (BelehovEgor) [inline src/RightsService.Validation/Helpers/RoleRightsCompareHelper.cs:18]: you can use select

File: test/RightsService.Data.UnitTests/RoleRepositoryTests.cs [hunk 5]
```diff
@@ -107,5 +108,16 @@ public void ShouldCreateRole()
         }
 
         #endregion
+
+        #region GetAll
+
+        [Test]
+        public void SholdGetAllRoles()
```
Comment (BelehovEgor) [inline test/RightsService.Data.UnitTests/RoleRepositoryTests.cs:115]: typo should

File: test/RightsService.Validation.UnitTests/RightsService.Validation.UnitTests.csproj [hunk 6]
```diff
@@ -2,6 +2,10 @@
 
   <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
 
+  <ItemGroup>
```
Comment (BelehovEgor) [inline test/RightsService.Validation.UnitTests/RightsService.Validation.UnitTests.csproj:5]: delete this and update unit test kernel packet in solution items

**Review by BelehovEgor — APPROVED** [review]

**Review by VLiamin — APPROVED** [review]

File: src/RightsService.Validation/Helpers/Interfaces/IRoleRightsCompareHelper.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+using LT.DigitalOffice.RightsService.Data.Interfaces;
+using System;
+using System.Collections.Generic;
+
+namespace LT.DigitalOffice.RightsService.Validation.Helpers.Interfaces
+{
+    [AutoInject]
+    public interface IRoleRightsCompareHelper
+    {
+        public bool Compare(List<int> addedRights);
```
Comment (BelehovEgor) [inline src/RightsService.Validation/Helpers/Interfaces/IRoleRightsCompareHelper.cs:11]: remove public

**Review by BelehovEgor — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

**Pull request closed without merging**